app/views/audits/index.html.erb

```diff
@@ -15,18 +15,17 @@
     </tr>
   </thead>
   <tbody class="govuk-table__body">
-    <tr class="govuk-table__row">
-      <td class="govuk-table__cell"> <%= @planning_application.created_at.strftime("%d-%m-%Y %H:%M") %></td>
-      <td class="govuk-table__cell"> API </td>
-      <td class="govuk-table__cell"> Application created </td>
-    </tr>
     <% @audits.each do |audit_item| %>
       <tr class="govuk-table__row">
         <td class="govuk-table__cell"><%= audit_item.created_at.strftime("%d-%m-%Y %H:%M") %></td>
-        <td class="govuk-table__cell"><%= audit_item.user.name %></td>
+        <% if audit_item.user %>
+          <td class="govuk-table__cell"><%= audit_item.user.name %></td>
```

Comment:
One last thing: what happens is a user or api_user gets deleted? I think it would cause an error on the audit log page? Two ways of resolving this:

1) we set `has_many :audits, dependent: :destroy` on User and ApiUser, which will ensure that if we delete a user, the audit logs get deleted to
2) we set `has_many :audits, dependent: :nullify` on User and ApiUser, and ensure that on this line we catch if `audit_item.user` is nil, we say "User deleted" instead.

I think the second option is better?

Reply:
I tried deleting it and nothing much happened:

![image](https://user-images.githubusercontent.com/9452321/111162190-d7ac1d80-8593-11eb-8b89-20b485e3cf26.png)

will see if dependent nullify keeps the record whilst having an empty user, probably that's indeed the best way to go

Reply:
so because I did not enforce any has_many (we only have a belongs_to on the api, and it has "optional true") it was enough to just add an "else" statement to have the user as deleted

![image](https://user-images.githubusercontent.com/9452321/111165010-7c2f5f00-8596-11eb-876a-3f2929c219e9.png)

```
      <td class="govuk-table__cell"><%= audit_item.created_at.strftime("%d-%m-%Y %H:%M") %></td>
        <% if audit_item.user %>
          <td class="govuk-table__cell"><%= audit_item.user.name %></td>
        <% elsif audit_item.api_user %>
          <td class="govuk-table__cell"><%= audit_item.api_user.name %></td>
        <% else %>
          <td class="govuk-table__cell"> User deleted </td>
        <% end %>
       ``` 
       
       this seemed like the "minimum effort" approach. Since the users are always marked as optional on the model I didn't see any possible problems with just doing this little catch clause.. sounds ok?

Reply:
That works well :-)

I'd still like to have the has_many dependent nullify, as that way we won't be left with records that have a user_id/api_user_id that points to a deleted record, if that makes sense.

Reply:
Not that that will actually prevent any errors, but might help us out if we're trying to debug something in a hurry later on.